### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ntfshard**

### CODE REVIEW

**Comment by ntfshard** [comment]
make test

File: include/data_management/data/data_block.h [hunk 1]
```diff
@@ -109,13 +109,9 @@ class DAAL_EXPORT DataBlock : public DataBlockIface
...
      */
-    DataBlock(const DataBlock & block)
-    {
-        _ptr  = block._ptr;
-        _size = block._size;
-    }
+    DataBlock(const DataBlock & block) : _ptr(block._ptr), _size(block._size) {}
 
-    virtual ~DataBlock() {}
+    virtual ~DataBlock() DAAL_C11_OVERRIDE {}
```
Comment (mchernov-intel) [inline include/data_management/data/data_block.h]: delete virtual

File: include/data_management/data_source/modifiers/csv/internal/default_modifiers.h [hunk 2]
```diff
@@ -156,10 +156,12 @@ class CategoricalFeatureModifier : public FeatureModifier
...
-
-        for (size_t i = 0; i < numberOfInputFeatures; i++)
+        else
         {
-            _primitives[i].initialize(config, i);
+            for (size_t i = 0; i < numberOfInputFeatures; i++)
+            {
+                _primitives[i].initialize(config, i);
+            }
         }
```
Comment (mchernov-intel) [inline include/data_management/data_source/modifiers/csv/internal/default_modifiers.h]: revert this change, please

**Comment by ntfshard** [comment]
make test

**Comment by ntfshard** [comment]
restart
